tool/lib/license_utils.dart

```diff
@@ -109,90 +200,98 @@ class LicenseHeader {
       }
       // Return a non-empty map for the case where there is a stored value
       // requested (i.e. when there is a '<value>' defined in the license text)
-      final String storedName = _parseStoredName(existingLicenseText);
+      final storedName = _parseStoredName(existingLicenseText);
       if (storedName.isNotEmpty) {
         return _processHeaders(
-          storedName,
-          existingLicenseText,
-          replacementLicenseText,
-          content,
+          storedName: storedName,
+          existingLicenseText: existingLicenseText,
+          replacementLicenseText: replacementLicenseText,
+          content: content,
         );
       }
     }
     throw Exception('License header expected in ${file.path}, but not found!');
   }
 
   /// Returns a copy of the given [file] with the [existingHeader] replaced by
-  /// the [replacementHeader]. Reads and writes the entire file contents all
-  /// at once, so performance may degrade for large files.
-  Future<File> rewriteLicenseHeader(
-    File file,
-    String existingHeader,
-    String replacementHeader,
-  ) async {
-    final File rewrittenFile = File('${file.path}.tmp');
-    // TODO: [mossmana] Convert to using a stream
-    await file.readAsString().then((String contents) async {
-      final String replacementContents = contents.replaceFirst(
-        existingHeader,
-        replacementHeader,
-      );
-      await rewrittenFile.writeAsString(replacementContents);
-    });
+  /// the [replacementHeader].
+  ///
+  /// Reads and writes the entire file contents all at once, so performance may
+  ///  degrade for large files.
+  File rewriteLicenseHeader({
+    required File file,
+    required String existingHeader,
+    required String replacementHeader,
+  }) {
+    final rewrittenFile = File('${file.path}.tmp');
+    // TODO(mossmana): Convert to using a stream
+    final contents = file.readAsStringSync();
+    final replacementContents = contents.replaceFirst(
+      existingHeader,
+      replacementHeader,
+    );
+    rewrittenFile.writeAsStringSync(replacementContents, flush: true);
     return rewrittenFile;
   }
 
   /// Bulk update license headers for files in the [directory] as configured
-  /// in the [config]. Returns a list of file paths that were updated.
-  Future<Map<String, List<String>>> bulkUpdate(
-      Directory directory, LicenseConfig config,) async {
-    final List<String> includedPaths = [];
-    final List<String> updatedPaths = [];
-    final List<File> files =
+  /// in the [config] and return a list of file paths that were updated.
+  ///
+  /// If [dryRun] is set, returns a list of file paths that should be updated,
+  /// but no files will be actually be updated.
+  Future<Map<String, List<String>>> bulkUpdate({
```

Comment:
It's not clear just by looking at this return type and reading the documentation what the format of this returned map is. Consider making the documentation more explicit or make use of typedefs to make it more clear what the keys and values in the map represent.

Reply:
@bkonyi I switched to a typedef and tried to make the docs a bit clearer. Let me know if there's anything else I should add.

Reply:
This is still the raw type :)

Reply:
There were a few places where a typedef would have been useful. I accidentally fixed one of the other instances but not this one. Should be good now.